### PR TITLE
feat: STRF-9741 Verbose error logging in Stencil CLI

### PR DESCRIPTION
**Pull request by jairo-bc**
#### What?

Introduced verbose network requests logging.

#### Tickets / Documentation

-   [STRF-9741](https://jira.bigcommerce.com/browse/STRF-9741)

#### Screenshots (if appropriate)

Screenshot 2022-04-19 at 13 29 09
Screenshot 2022-04-19 at 13 29 22
Screenshot 2022-04-19 at 13 29 46
Screenshot 2022-04-19 at 13 30 05

cc @bigcommerce/storefront-team

### CODE REVIEW

File: lib/StencilCLISettings.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        // The setting enables/disables verbose network requests logging
+        this.versboseNetworkLogging = true;
+    }
+
+    setVerbose(flag) {
+        this.versboseNetworkLogging = flag;
+    }
+
+    isVerbose() {
+        return this.versboseNetworkLogging === true;
```
Comment (bc-max) [inline lib/StencilCLISettings.js:12]: Cannot we just do as follows?
```suggestion
        return this.versboseNetworkLogging;
```

**Review by bc-max — APPROVED** [review]

File: lib/StencilCLISettings.js (hunk 1 above)
Comment (jairo-bc) [reply lib/StencilCLISettings.js:12]: @bc-max yeah, we can

File: lib/cliCommon.js [hunk 2]
```diff
@@ -8,6 +9,33 @@ const messages = {
...
+/**
+ * @param {Error} error
+
+ * @returns {void}
+ */
+function printNetworkError(config) {
+    console.log(`URL: `.yellow + config.url);
+    console.log(`Method: `.yellow + config.method.toUpperCase());
+    console.log(`Headers:`.yellow);
+    printObject(config.headers);
```
Comment (zvuki) [inline lib/cliCommon.js:32]: let's not print headers, as they may contain auth tokens

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 ### Draft
 
+-   feat: STRF-9741 Verbose error logging in Stencil CLI by default ([x](https://github.com/bigcommerce/stencil-cli/pull/x))
```
Comment (zvuki) [inline CHANGELOG.md:3]: can you add a line on how to disable it?

**Comment by zvuki** [comment]
@jairo-bc Do you have an example of the output when an error happens?

File: CHANGELOG.md (hunk 3 above)
Comment (jairo-bc) [reply CHANGELOG.md:3]: sure

**Comment by jairo-bc** [comment]
Screenshot 2022-04-20 at 20 58 06
Screenshot 2022-04-20 at 21 01 36

**Review by zvuki — APPROVED** [review]

**Pull request merged**